contrib/lkt/extensions/mains/lkt_toolbox.adb

```diff
@@ -42,13 +45,37 @@ procedure Lkt_Toolbox is
 
    use Liblktlang;
 
+   package Invalid_Decl_Maps is new Ada.Containers.Hashed_Maps
+     (Key_Type        => Analysis.LK_Node,
+      Element_Type    => Boolean,
+      Hash            => Analysis.Hash,
+      Equivalent_Keys => "=");
+
    procedure Print_Semantic_Result
      (S : Analysis.Semantic_Result; Unit : Analysis.Analysis_Unit);
    --  Print a semantic result
 
    function Format_Node (Decl_Node : Decl'Class) return String;
    --  Format node for semantic result printing
 
+   procedure Print_Lkt_Toolbox_Diagnostic
+     (Node : LK_Node'Class; Message : Unbounded_Text_Type);
```

Comment:
It looks like all calls to this procedure use `To_Unbounded_Text`: what about making it part of the factorization too?

Reply:
Done!